### PR TITLE
Documentation cleanup

### PR DESCRIPTION
**Pull request by josegonzalez**
Starting on this! Feel free to tack on examples or more docs. I'm hoping to document every plugin, and maybe show examples for each.

### CODE REVIEW

File: docs/pluginhooks.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+- pluginhook post-deploy  $APP $port $ipaddr
+- pluginhook pre-delete $APP
+- pluginhook post-delete $APP
+- pluginhook docker-args $APP build
+- pluginhook docker-args $APP deploy
+- pluginhook docker-args $APP run
+- pluginhook bind-external-ip $APP
+- pluginhook post-domains-update $APP
+- pluginhook post-domains-update $APP
+- pluginhook post-domains-update $APP
```
Comment (vincentfretin) [inline docs/pluginhooks.md]: @josegonzalez You have three times "pluginhook post-domains-update $APP" here.

File: docs/pluginhooks.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+- pluginhook pre-release $APP
+- pluginhook post-release $APP
+- pluginhook check-deploy $id $APP $port ${ipaddr:-localhost}
+- pluginhook pre-deploy $APP
+- pluginhook post-deploy  $APP $port $ipaddr
+- pluginhook pre-delete $APP
+- pluginhook post-delete $APP
+- pluginhook docker-args $APP build
+- pluginhook docker-args $APP deploy
+- pluginhook docker-args $APP run
```
Comment (michaelshobbs) [inline docs/pluginhooks.md]: We may want to talk about reasons we're passing in the phase name for the docker-args pluginhook. Additionally, is it interesting to pass this into all pluginhooks?

Comment (josegonzalez) [inline docs/pluginhooks.md]: The phase? Don't think so, as other pluginhooks already state which phase they are in.

File: docs/pluginhooks.md (hunk 1 above)
Comment (josegonzalez) [inline docs/pluginhooks.md]: Good catch, I'm not done with this PR though :)

File: docs/pluginhooks.md (hunk 2 above)
Comment (michaelshobbs) [inline docs/pluginhooks.md]: That's my derp there. Still recovering from illness over the weekend.

**Comment by michaelshobbs** [comment]
We'll want to include `docker-args-PHASE` docs here.

ref: #927

**Comment by josegonzalez** [comment]
Ready for review. I can always write more docs, though this is largely a restructuring.

**Comment by josegonzalez** [comment]
I've included all the changes from this PR in other docs PRs I made - and merged - today, so alles ist guht :)

**Pull request closed without merging**